lib/elixir/test/elixir/macro_test.exs

```diff
@@ -620,6 +621,57 @@ defmodule MacroTest do
     assert Macro.unpipe(quote(do: foo |> bar |> baz)) == quote(do: [{foo, 0}, {bar, 0}, {baz, 0}])
   end
 
+  # guard
+
+  test "guard" do
```

Comment:
I meant to ask about that--I'm struggling to catch those syntatic `ArgumentErrors` and semantic `CompileErrors` in ExUnit; neither `catch_error` or `assert_error` are up to the task since the errors are compile-time.